### PR TITLE
Rsdk 1831 slam wrapper

### PR DESCRIPTION
**Pull request by purplenicole730**
From now on, service classes in `server.py` files are `ResourceRPCService`, including in components. Service clients now don't have the word `Service` between the `Resource` and `Client`.

This is to help prepare as we move towards supporting `services`.

### CODE REVIEW

File: tests/test_slam_service.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    GetPositionResponse,
+    SLAMServiceStub,
+)
+from viam.resource.manager import ResourceManager
+from viam.services.slam import Pose, SLAMServiceClient, SLAMServiceRPCService
+from viam.utils import dict_to_struct, struct_to_dict
+
+from .mocks.services import MockSLAMService
+
+INTERNAL_STATE_CHUNKS = [bytes(5), bytes(2)]
```
Comment (stuqdog) [inline tests/test_slam_service.py:22]: (q) There's some repetition here with `tests/mocks/services.py` and `examples/servers/v1/services.py`. I assume there's a good reason for the duplication at least as far as the example code goes, but is it possible/would it make sense to dedup in the `tests/` code?

**Review by stuqdog — APPROVED** [review]
LGTM, modulo one question!

File: tests/test_slam_service.py (hunk 1 above)
Comment (purplenicole730) [reply tests/test_slam_service.py:22]: I know that they're used for different things. I followed the way we do components for services. We also do the same for `modules`. As for why we split them this way, I'm not entirely sure.

File: src/viam/services/slam/slam.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import abc
+from typing import Final, List, Optional
+
+from viam.resource.types import RESOURCE_NAMESPACE_RDK, RESOURCE_TYPE_SERVICE, Subtype
+
+from ..service_base import ServiceBase
+from . import Pose
+
+
+class SLAMService(ServiceBase):
```
Comment (cheukt) [inline src/viam/services/slam/slam.py:10]: just SLAM is fine

File: src/viam/services/slam/service.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        timeout = stream.deadline.time_remaining() if stream.deadline else None
+        position = await slam.get_position(timeout=timeout)
+        response = GetPositionResponse(pose=position)
+        await stream.send_message(response)
+
+    async def DoCommand(self, stream: Stream[DoCommandRequest, DoCommandResponse]) -> None:
+        request = await stream.recv_message()
+        assert request is not None
+        try:
+            arm = self.get_resource(request.name)
```
Comment (cheukt) [inline src/viam/services/slam/service.py:72]: arm -> slam

File: src/viam/services/slam/slam.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    This acts as an abstract base class for any drivers representing specific
+    arm implementations. This cannot be used on its own. If the ``__init__()`` function is
+    overridden, it must call the ``super().__init__()`` function.
+    """
+
+    SUBTYPE: Final = Subtype(RESOURCE_NAMESPACE_RDK, RESOURCE_TYPE_SERVICE, "slam")
+
+    @abc.abstractmethod
+    async def get_internal_state(self, *, timeout: Optional[float]) -> List[bytes]:
+        """Get the internal state of the SLAM algorithm required to continue mapping/localization
```
Comment (cheukt) [inline src/viam/services/slam/slam.py:23]: nit: docs should end with periods

**Review by cheukt — CHANGES_REQUESTED** [review]
small nits, but generally looks great!

File: src/viam/services/slam/service.py (hunk 3 above)
Comment (purplenicole730) [reply src/viam/services/slam/service.py:72]: Thanks for catching this! Haha

File: src/viam/rpc/server.py [hunk 5]
```diff
@@ -29,7 +29,7 @@ class Server(ResourceManager):
     gRPC Server
     """
 
-    def __init__(self, components: List[ResourceBase], *, module_service: Optional["ModuleService"] = None):
+    def __init__(self, components: List[ResourceBase], *, module_service: Optional["ModuleRPCService"] = None):
```
Comment (cheukt) [inline src/viam/rpc/server.py:32]: rename this to resources and then update the docstring

File: tests/test_slam_service.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+
+class TestSLAMService:
+    name = "slam"
+    slam = MockSLAM(name="slam")
+
+    @pytest.mark.asyncio
+    async def test_get_internal_state_chunks(self):
+        chunks = await self.slam.get_internal_state()
+        assert chunks == INTERNAL_STATE_CHUNKS
```
Comment (cheukt) [inline tests/test_slam_service.py:34]: can now assert against MockSLAM.INTERNAL_STATE_CHUNKS

File: tests/test_slam_service.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
+from typing import List
```
Comment (cheukt) [inline tests/test_slam_service.py:1]: rename file to `test_slam`

**Review by cheukt — APPROVED** [review]

**Pull request merged**